sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -309,6 +310,18 @@ public static GenericRecord toGenericRecord(
     return g -> toGenericRecord(g, avroSchema);
   }
 
+  /** Transform an existing Avro-based PCollection input into an Schema-based PCollection. */
+  public static <T> PCollection<T> asSchemaPCollection(
+      PCollection<T> pc, Class<T> clazz, @Nullable org.apache.avro.Schema schema) {
+    if (!pc.hasSchema()) {
+      Schema beamSchema = getSchema(clazz, schema);
+      if (beamSchema != null) {
+        pc.setSchema(beamSchema, getToRowFunction(clazz, schema), getFromRowFunction(clazz));
```

Comment:
`pc.setSchema` is equivalent to `pc.setCoder(SchemaCoder.of(...))`

I was thinking, what if we just create a static method to create `SchemaCoder` instead, then user-facing API would be like:
```
KafkaIO.read(MyRecord.class)
  .setCoder(AvroUtils.schemaCoder(MyRecord.class))
```

or, it can be done implicitly by registering classes in advance:

```
p.getSchemaRegistry().registerSchemaProvider(MyRecord.class, new AvroRecordSchema());
```

or for every `SpecificRecord` (didn't try this, but it should work as well):
```
p.getSchemaRegistry().registerSchemaProvider(SpecificRecord.class, new AvroRecordSchema());
```

Reply:
Yes good idea, that looks cleaner I will add the `schemaCoder` method to AvroUtils, it should cover all the mentioned cases. will ping you back when done. Thanks for the review ideas.